docs/application/dotnet/guides/machine-learning/pipeline.md

```diff
@@ -0,0 +1,180 @@
+
+```C#
+/* Get the source node with the name 'srcx'. */
+var src_node = pipe.GetSource("srcx");
+
+/* Start the pipeline. */
+pipe.Start();
+
+/* Prepare input tensor data, with data type UInt8 and dimension 3:224:224:1. */
+var in_info = new TensorsInfo();
+in_info.AddTensorInfo(TensorType.UInt8, new int[4] { 3, 224, 224, 1 });
+
+var in_data = in_info.GetTensorsData();
+
+/* Push input tensor data into the pipeline. */
+src_node.Input(in_data);
+```
+
+<a name="outputdata"></a>
+## Subscribing to the output data
+
+You can use `SinkNode` to get the output tensor data from the pipeline:
+
+```C#
+/* Firstly, declare the event handler for the data received event. */
+void DataReceivedEvent(object sender, DataReceivedEventArgs args)
+{
+    /* Add your task with received data. */
+    TensorsData dataReceived = args.Data;
+}
+```
+
+```C#
+/* Get the sink node with the name 'sinkx'. */
+var sink_node = pipe.GetSink("sinkx");
+
+/* Subscribe the event to get the data from the sink node. */
+sink_node.DataReceived += DataReceivedEvent;
+
+/* Start the pipeline. */
+pipe.Start();
+```
+
+<a name="dataflow"></a>
+## Data flow control
+
+You need to stop the data stream or select the data flow with multiple stream paths for neural network models. Pipeline provides methods to handle the flow.
+
+1. Start or stop the data flow.
+
+    `Start()` and `Stop()` control the overall data flow of the pipeline asynchronously.
+    You can also get the pipeline state from the `State` property:
+
+    ```C#
+    string description = "input-selector name=ins ! tensor_converter ! valve name=valvex ! tensor_sink name=sinkx " +
+                         "videotestsrc is-live=true ! videoconvert ! ins.sink_0 " +
+                         "videotestsrc is-live=true ! videoconvert ! ins.sink_1";
+    var pipe = new Pipeline(description);
+
+    /* After constructing the pipeline, the pipeline state is 'Paused'. */
+    var pipeline_state = pipe.State;
+
+    /* Start the pipeline (The pipeline state will be 'Playing'). */
+    pipe.Start();
+
+    /* Stop the pipeline (The pipeline state will be 'Paused'). */
+    pipe.Stop();
+    ```
+
+2. Open or close the stream.
+
+    The 'valve' element controls the stream of a pipeline.
+    When including 'valve' in pipeline description, you can get the instance of `ValveNode` with its name.
+    It provides a method to let the flow pass to a downstream element, or stop the flow:
+
+    ```C#
+    /* Get the valve node with the name 'valvex'. */
+    var valve_node = pipe.GetValve("valvex");
+
+    /* Close valve and stop the flow. */
+    valve_node.Control(false);
+
+    /* Start the pipeline (The pipeline state will be PLAYING). The sink node with the name 'sinkx' cannot receive the data. */
+    pipe.Start();
+    ```
+
+3. Select the stream path.
+
+    The 'input-selector' and 'output-selector' are the elements to select the data flow with multiple stream paths.
```

Comment:
Thank you for updating the sentence.
Please remove "the" before "input-selector".
We can use the code tags instead of using quotes (' '). For more information, see https://github.com/Samsung/tizen-docs/pull/1079#discussion_r408586270.

`input-selector` and `output-selector` are the elements to select the data flow with multiple stream paths.

Please verify.